packages/core/src/external/cda/partition-payload.ts

```diff
@@ -27,22 +27,24 @@ export function partitionPayload(payloadRaw: string): string[] {
   }
 
   const chunks: string[] = [];
+
   let groupedComponents: string[] = [];
   let groupedSize = 0;
-
   const components = toArray(json.ClinicalDocument?.component?.structuredBody?.component);
   for (const currentComponent of components) {
     const currentSize = sizeInBytes(JSON.stringify(currentComponent));
 
-    if (currentSize > MAX_CHUNK_SIZE_IN_BYTES) {
-      chunks.push(createChunk(json, [currentComponent]));
-      continue;
-    }
-
     if (groupedSize + currentSize > MAX_CHUNK_SIZE_IN_BYTES) {
-      chunks.push(createChunk(json, [groupedComponents, currentComponent]));
-      groupedComponents = [];
-      groupedSize = 0;
+      if (groupedComponents.length > 0) {
+        chunks.push(createChunk(json, groupedComponents));
+        groupedComponents = [currentComponent];
+        groupedSize = currentSize;
+      } else {
+        // currentComponent size is larger than max chunk size
```

Comment:
nit: this is only true for the first item... from the second element onwards, if the current component is larger than max chunk size, it will execute the `if`, not the `else`, right?

Reply:
yea, that's why the comment is in the `else`

Reply:
I mean the `else` only gets executed with the first item, so the current comment is not very accurate as when "currentComponent size is larger than max chunk size" and "not the first item" it won't execute that part of the code.

That's why I had it "first element is larger than max (chunk size)"

No biggie, just to avoid a comment that could be confusing. Could just remove the comment altogether.